### PR TITLE
Manage JVM vs. CLJS dependencies better

### LINKED ISSUE #42: clj-time is deprecated; README should use Java Time instead
It's fine for `humanize` to continue using `clj-time` for its implementation but the README really should be updated to use Java Time instead, so that we don't continue to perpetuate a long-deprecated date/time library.

### PR DESCRIPTION
**Pull request by seancorfield**
Fixes #42 by:

* use Java Time in README examples
* removes clj-time as a dependency
* moves js-joda to a :cljs-test only dependency
* kongeor/cljs-test-runner is behind olical/cljs-test-runner so switch to the latter
* update build-tools
* update GH workflow dependencies
* ignore Calva REPL file
* remove unnecessary cljc_test file (both clj and cljs test runners file the tests just fine)
* use clojure.math in cljs and clj
* fix the bad cljs-only test splice

Signed-off-by: Sean Corfield <sean@corfield.org>

### CODE REVIEW

**Comment by seancorfield** [comment]
The `cljc_test.cljc` file caused errors when `clojure -M:cljs-test` was run (`window` not defined) and it really doesn't add much to the Chrome/Firefox test output on Karma (I've never seen anyone bother to test that way, beyond play `clojure -M:cljs-test`).

File: README.md [hunk 1]
```diff
@@ -197,19 +197,19 @@ Given a datetime or date, return a human-friendly representation
 of the amount of time difference, relative to the current time.
 
 ```clojure
-user> (require '[clj-time.core :as t])
-nil
+user> (import '(java.time LocalDateTime) '(java.time.temporal ChronoUnit))
+java.time.temporal.ChronoUnit
```
Comment (kongeor) [inline README.md:201]: Probably redundant?

**Review by kongeor — APPROVED** [review]
Thank you Sean. Looks good!

File: README.md (hunk 1 above)
Comment (seancorfield) [reply README.md:201]: Both of these imports are used so I'm not sure why you think this is redundant?

**Comment by hlship** [comment]
Thanks for this; sorting out the JVM vs. CLJS stuff was something outside my wheelhouse as I don't do much with CLJS.

**Pull request merged**